executor/executor.go

```diff
@@ -237,12 +262,13 @@ func Run(path string, env []string, emitter screwdriver.Emitter, build screwdriv
 			"EXITCODE=$?; " +
 			exportEnvCmd +
 			"echo $SD_STEP_ID $EXITCODE; }", //mv newfile to file
-		"trap finish EXIT;\n",
+		"trap finish EXIT;\necho ;\n",
 	}
 
-	shargs := strings.Join(setupCommands, " && ")
-
-	f.Write([]byte(shargs))
+	setupReader := bufio.NewReader(f)
+	if err := doRunSetupCommand(emitter, f, setupReader, setupCommands); err != nil {
+		return err
```

Comment:
How about creating more precise error message (kind of showing setup command failure) at here instead of inside `doRunSetupCommand` ?

Reply:
I fixed the doRunSetupCommand and it now returns several types of errors. That's why we leave the error message to be created inside doRunSetupCommand.

Reply:
I see. It looks reasonable.